core/src/main/java/io/kestra/plugin/core/kv/PurgeKV.java

```diff
@@ -0,0 +1,186 @@
+                    includeChildNamespaces: true # default true
+                    keyPattern: * # by default all, optionally specify Glob-Pattern e.g. AI_*
+                """
+        )
+    }
+)
+public class PurgeKV extends Task implements RunnableTask<PurgeKV.Output> {
+
+    @Schema(
+        title = "Key pattern. Delete only key matching the blob pattern"
+    )
+    private Property<String> keyPattern;
+
+    @Schema(
+        title = "List of namespaces to delete key from"
+    )
+    private Property<List<String>> namespaces;
+
+    @Schema(
+        title = "Blob pattern fo the namespaces to delete key from"
+    )
+    private Property<String> namespacePattern;
+
+    @Schema(
+        title = "Delete only expired keys. Default true"
+    )
+    @Builder.Default
+    private Property<Boolean> expiredOnly = Property.ofValue(true);
+
+    @Schema(
+        title = "Delete keys from child namespaces"
+    )
+    @Builder.Default
+    private Property<Boolean> includeChildNamespaces = Property.ofValue(true);
+
+
+    @Override
+    public Output run(RunContext runContext) throws Exception {
+        List<String> kvNamespaces = findNamespaces(runContext);
+        boolean expired = runContext.render(expiredOnly).as(Boolean.class).orElse(true);
+        String renderedKeyPattern = runContext.render(keyPattern).as(String.class).orElse(null);
+        boolean keyFiltering = StringUtils.isNotBlank(renderedKeyPattern);
+        AtomicLong count = new AtomicLong();
+        for (String ns : kvNamespaces) {
+            KVStore kvStore = runContext.namespaceKv(ns);
+            List<KVEntry> kvEntries = new ArrayList<>();
+            if (expired){
+                kvEntries.addAll(kvStore.listExpired());
+            } else {
+                kvEntries.addAll(kvStore.listAll());
+            }
+            List<String> keys = kvEntries.stream()
+                .map(KVEntry::key)
+                .filter(key -> {
+                    if (keyFiltering) {
+                        return FilenameUtils.wildcardMatch(key, renderedKeyPattern);
+                    }
+                    return true;
+                })
+                .toList();
+            for (String key : keys) {
+                kvStore.delete(key);
+            }
+            count.addAndGet(keys.size());
+        }
+
+        return Output.builder()
+            .size(count.get())
+            .build();
+    }
+
+    @VisibleForTesting
+    protected List<String> findNamespaces(RunContext runContext) throws IllegalVariableEvaluationException {
+        String tenantId = runContext.flowInfo().tenantId();
+        String currentNamespace = runContext.flowInfo().namespace();
+        FlowRepositoryInterface flowRepositoryInterface = ((DefaultRunContext) runContext)
+            .getApplicationContext().getBean(FlowRepositoryInterface.class);
+        List<String> distinctNamespaces = flowRepositoryInterface.findDistinctNamespace(tenantId);
```

Comment:
Accessing a repository for a task is a bad pattern that would soon be prohibited.
Doing so creates technical debt